src/api/firebase.js

```diff
@@ -71,15 +78,30 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
+		checked: false,
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+/**
+ * update a user's list item in Firebase
+ * @param {*} listId The list token we are modifying its data.
+ * @param {*} itemId The item id.
+ * @param {*} checked A boolean value indicating whether the item is purchased.
+ */
+export async function updateItem(listId, itemId, checked) {
+	const itemRef = doc(db, listId, itemId);
+	const itemDoc = await getDoc(itemRef);
+	const { dateLastPurchased, totalPurchases } = itemDoc.data();
```

Comment:
Re: line 93
I wonder if would be safer to also add a check  here to see if itemDoc actually exists?

Reply:
We also thought about it, but in this case since the `updateItem` function are taking `listToken` and item `data` input directly from App.jsx, they should always be valid, so we didn't think it's necessary to add.

Reply:
I was thinking of a scenario (and not just for this issue, but in general) where two users share a list, and at the same moment (I know, what are the odds of that!) when one is trying to access an item, the other user is deleting the same item. 🤔 The `useShoppinListData` function has a `useEffect` that is supposed the manage these type of changes to the data, but this is just me still not trusting that React could really be that smart 😂

Reply:
I like how you are thinking about these edge cases! In that case, the change should be reflected instantly on the screen because `useShoppingListData` retrieves a real-time update, so we can trust the process!

Reply:
Very thoughtful review and implementation here @CorinaMurg and @shangguanwang !! 👏